### ISSUE TITLE
Proxy registry crashes at start when there is no Internet connection

### ISSUE THREAD

**Issue by mikef-phy** [distribution/distribution#4038]
Need to be able to use docker proxy if internet connectivity is lost.  The server will not start and host cached images when internet is not available on boot.

goroutine 1 [running]:
github.com/docker/distribution/registry/handlers.NewApp({0xe5e8a0?, 0xc0003fa960?}, 0xc0003fc000)
	github.com/docker/distribution/registry/handlers/app.go:324 +0x1c0d
github.com/docker/distribution/registry.NewRegistry({0xe5e8a0?, 0xc0003c5f80?}, 0xc0003fc000)
	github.com/docker/distribution/registry/registry.go:161 +0x145
github.com/docker/distribution/registry.glob..func1(0xc000092170?, {0xc0002f8c00, 0x1, 0x1})
	github.com/docker/distribution/registry/registry.go:119 +0x18f
github.com/docker/distribution/vendor/github.com/spf13/cobra.(*Command).execute(0x1369f60, {0xc0002f8bc0?, 0x2?, 0x2?})
	github.com/docker/distribution/vendor/github.com/spf13/cobra/command.go:495 +0x44f
github.com/docker/distribution/vendor/github.com/spf13/cobra.(*Command).Execute(0x136a100)
	github.com/docker/distribution/vendor/github.com/spf13/cobra/command.go:560 +0xaa
main.main()
	github.com/docker/distribution/cmd/registry/main.go:23 +0x25
time="2023-09-04T13:44:58.243537226Z" level=warning msg="No HTTP secret provided - generated random secret. This may cause problems with uploads if multiple registries are behind a load-balancer. To provide a shared secret, fill in http.secret in the configuration file or set the REGISTRY_HTTP_SECRET environment variable." go.version=go1.19.9 instance.id=22602e53-3aa6-4f80-9cf3-522b51ac5fd0 service=registry version=2.8.2 
time="2023-09-04T13:44:58.243682676Z" level=info msg="Starting upload purge in 17m0s" go.version=go1.19.9 instance.id=22602e53-3aa6-4f80-9cf3-522b51ac5fd0 service=registry version=2.8.2 
time="2023-09-04T13:44:58.244019034Z" level=info msg="redis not configured" go.version=go1.19.9 instance.id=22602e53-3aa6-4f80-9cf3-522b51ac5fd0 service=registry version=2.8.2 
time="2023-09-04T13:44:58.244304642Z" level=info msg="using inmemory blob descriptor cache" go.version=go1.19.9 instance.id=22602e53-3aa6-4f80-9cf3-522b51ac5fd0 service=registry version=2.8.2 
time="2023-09-04T13:44:58.251429661Z" level=info msg="Starting cached object TTL expiration scheduler..." go.version=go1.19.9 instance.id=22602e53-3aa6-4f80-9cf3-522b51ac5fd0 service=registry version=2.8.2 
panic: Get "https://registry-1.docker.io/v2/": dial tcp: lookup registry-1.docker.io on 10.1.3.1:53: server misbehaving

goroutine 1 [running]:
github.com/docker/distribution/registry/handlers.NewApp({0xe5e8a0?, 0xc0003f8b70?}, 0xc0003b9880)
	github.com/docker/distribution/registry/handlers/app.go:324 +0x1c0d
github.com/docker/distribution/registry.NewRegistry({0xe5e8a0?, 0xc0003f81b0?}, 0xc0003b9880)
	github.com/docker/distribution/registry/registry.go:161 +0x145
github.com/docker/distribution/registry.glob..func1(0xc0000341a0?, {0xc0003096e0, 0x1, 0x1})
	github.com/docker/distribution/registry/registry.go:119 +0x18f
github.com/docker/distribution/vendor/github.com/spf13/cobra.(*Command).execute(0x1369f60, {0xc0003096a0?, 0x2?, 0x2?})
	github.com/docker/distribution/vendor/github.com/spf13/cobra/command.go:495 +0x44f
github.com/docker/distribution/vendor/github.com/spf13/cobra.(*Command).Execute(0x136a100)
	github.com/docker/distribution/vendor/github.com/spf13/cobra/command.go:560 +0xaa
main.main()
	github.com/docker/distribution/cmd/registry/main.go:23 +0x25
time="2023-09-04T13:45:50.365608691Z" level=warning msg="No HTTP secret provided - generated random secret. This may cause problems with uploads if multiple registries are behind a load-balancer. To provide a shared secret, fill in http.secret in the configuration file or set the REGISTRY_HTTP_SECRET environment variable." go.version=go1.19.9 instance.id=48510656-5470-4fc4-91fe-001413ab6c29 service=registry version=2.8.2 
time="2023-09-04T13:45:50.36574565Z" level=info msg="redis not configured" go.version=go1.19.9 instance.id=48510656-5470-4fc4-91fe-001413ab6c29 service=registry version=2.8.2 
time="2023-09-04T13:45:50.365800129Z" level=info msg="Starting upload purge in 29m0s" go.version=go1.19.9 instance.id=48510656-5470-4fc4-91fe-001413ab6c29 service=registry version=2.8.2 
time="2023-09-04T13:45:50.366012801Z" level=info msg="using inmemory blob descriptor cache" go.version=go1.19.9 instance.id=48510656-5470-4fc4-91fe-001413ab6c29 service=registry version=2.8.2 
time="2023-09-04T13:45:50.366264921Z" level=info msg="Starting cached object TTL expiration scheduler..." go.version=go1.19.9 instance.id=48510656-5470-4fc4-91fe-001413ab6c29 service=registry version=2.8.2 
panic: Get "https://registry-1.docker.io/v2/": dial tcp: lookup registry-1.docker.io on 10.1.3.1:53: server misbehaving

goroutine 1 [running]:
github.com/docker/distribution/registry/handlers.NewApp({0xe5e8a0?, 0xc0003cb380?}, 0xc0003b0a80)
	github.com/docker/distribution/registry/handlers/app.go:324 +0x1c0d
github.com/docker/distribution/registry.NewRegistry({0xe5e8a0?, 0xc0003ca9c0?}, 0xc0003b0a80)
	github.com/docker/distribution/registry/registry.go:161 +0x145
github.com/docker/distribution/registry.glob..func1(0xc0000341a0?, {0xc00034d2c0, 0x1, 0x1})
	github.com/docker/distribution/registry/registry.go:119 +0x18f
github.com/docker/distribution/vendor/github.com/spf13/cobra.(*Command).execute(0x1369f60, {0xc00034d280?, 0x2?, 0x2?})
	github.com/docker/distribution/vendor/github.com/spf13/cobra/command.go:495 +0x44f
github.com/docker/distribution/vendor/github.com/spf13/cobra.(*Command).Execute(0x136a100)
	github.com/docker/distribution/vendor/github.com/spf13/cobra/command.go:560 +0xaa
main.main()
	github.com/docker/distribution/cmd/registry/main.go:23 +0x25
time="2023-09-04T13:46:51.118164601Z" level=warning msg="No HTTP secret provided - generated random secret. This may cause problems with uploads if multiple registries are behind a load-balancer. To provide a shared secret, fill in http.secret in the configuration file or set the REGISTRY_HTTP_SECRET environment variable." go.version=go1.19.9 instance.id=234aab86-a040-41c0-b5e4-f3298e494772 service=registry version=2.8.2 
time="2023-09-04T13:46:51.11879609Z" level=info msg="redis not configured" go.version=go1.19.9 instance.id=234aab86-a040-41c0-b5e4-f3298e494772 service=registry version=2.8.2 
time="2023-09-04T13:46:51.119236Z" level=info msg="using inmemory blob descriptor cache" go.version=go1.19.9 instance.id=234aab86-a040-41c0-b5e4-f3298e494772 service=registry version=2.8.2 
time="2023-09-04T13:46:51.119742697Z" level=info msg="Starting cached object TTL expiration scheduler..." go.version=go1.19.9 instance.id=234aab86-a040-41c0-b5e4-f3298e494772 service=registry version=2.8.2 
time="2023-09-04T13:46:51.120185025Z" level=info msg="Starting upload purge in 16m0s" go.version=go1.19.9 instance.id=234aab86-a040-41c0-b5e4-f3298e494772 service=registry version=2.8.2 
panic: Get "https://registry-1.docker.io/v2/": dial tcp: lookup registry-1.docker.io on 10.1.3.1:53: server misbehaving

goroutine 1 [running]:
github.com/docker/distribution/registry/handlers.NewApp({0xe5e8a0?, 0xc0003f4b70?}, 0xc0003b7880)
	github.com/docker/distribution/registry/handlers/app.go:324 +0x1c0d
github.com/docker/distribution/registry.NewRegistry({0xe5e8a0?, 0xc0003f41b0?}, 0xc0003b7880)
	github.com/docker/distribution/registry/registry.go:161 +0x145
github.com/docker/distribution/registry.glob..func1(0xc0000341a0?, {0xc000083700, 0x1, 0x1})
	github.com/docker/distribution/registry/registry.go:119 +0x18f
github.com/docker/distribution/vendor/github.com/spf13/cobra.(*Command).execute(0x1369f60, {0xc0000836c0?, 0x2?, 0x2?})
	github.com/docker/distribution/vendor/github.com/spf13/cobra/command.go:495 +0x44f
github.com/docker/distribution/vendor/github.com/spf13/cobra.(*Command).Execute(0x136a100)
	github.com/docker/distribution/vendor/github.com/spf13/cobra/command.go:560 +0xaa
main.main()
	github.com/docker/distribution/cmd/registry/main.go:23 +0x25
root@dockerregistry:~# docker ps
CONTAINER ID        IMAGE               COMMAND                  CREATED             STATUS                         PORTS                  NAMES
a45af02878b2        codekoala/pypi      "/entrypoint.sh"         24 minutes ago      Up 3 minutes                   0.0.0.0:8080->80/tcp   docker_localpypi_1
d5de20eaa1dd        registry:latest     "/entrypoint.sh /etc…"   24 minutes ago      Restarting (2) 9 seconds ago

**Comment by milosgajdos** [comment]
Can you please list the version of the registry you are running and format your message? It makes it quite hard to parse useful info

**Comment by mikef-phy** [comment]
Latest.  And I'll format it.On Sep 4, 2023 9:57 AM, Milos Gajdos ***@***.***> wrote:
Can you please list the version of the registry you are running and format your message? It makes it quite hard to parse useful info

**Comment by mikef-phy** [comment]
Logs from container:

goroutine 1 [running]:
github.com/docker/distribution/registry/handlers.NewApp({0xe5e8a0?, 0xc0003fa960?}, 0xc0003fc000)
github.com/docker/distribution/registry/handlers/app.go:324 +0x1c0d
github.com/docker/distribution/registry.NewRegistry({0xe5e8a0?, 0xc0003c5f80?}, 0xc0003fc000)
github.com/docker/distribution/registry/registry.go:161 +0x145
github.com/docker/distribution/registry.glob..func1(0xc000092170?, {0xc0002f8c00, 0x1, 0x1})
github.com/docker/distribution/registry/registry.go:119 +0x18f
github.com/docker/distribution/vendor/github.com/spf13/cobra.(*Command).execute(0x1369f60, {0xc0002f8bc0?, 0x2?, 0x2?})
github.com/docker/distribution/vendor/github.com/spf13/cobra/command.go:495 +0x44f
github.com/docker/distribution/vendor/github.com/spf13/cobra.(*Command).Execute(0x136a100)
github.com/docker/distribution/vendor/github.com/spf13/cobra/command.go:560 +0xaa
main.main()
github.com/docker/distribution/cmd/registry/main.go:23 +0x25
time="2023-09-04T13:44:58.243537226Z" level=warning msg="No HTTP secret provided - generated random secret. This may cause problems with uploads if multiple registries are behind a load-balancer. To provide a shared secret, fill in http.secret in the configuration file or set the REGISTRY_HTTP_SECRET environment variable." go.version=go1.19.9 instance.id=22602e53-3aa6-4f80-9cf3-522b51ac5fd0 service=registry version=2.8.2
time="2023-09-04T13:44:58.243682676Z" level=info msg="Starting upload purge in 17m0s" go.version=go1.19.9 instance.id=22602e53-3aa6-4f80-9cf3-522b51ac5fd0 service=registry version=2.8.2
time="2023-09-04T13:44:58.244019034Z" level=info msg="redis not configured" go.version=go1.19.9 instance.id=22602e53-3aa6-4f80-9cf3-522b51ac5fd0 service=registry version=2.8.2
time="2023-09-04T13:44:58.244304642Z" level=info msg="using inmemory blob descriptor cache" go.version=go1.19.9 instance.id=22602e53-3aa6-4f80-9cf3-522b51ac5fd0 service=registry version=2.8.2
time="2023-09-04T13:44:58.251429661Z" level=info msg="Starting cached object TTL expiration scheduler..." go.version=go1.19.9 instance.id=22602e53-3aa6-4f80-9cf3-522b51ac5fd0 service=registry version=2.8.2
panic: Get "https://registry-1.docker.io/v2/": dial tcp: lookup registry-1.docker.io on 10.1.3.1:53: server misbehaving

goroutine 1 [running]:
github.com/docker/distribution/registry/handlers.NewApp({0xe5e8a0?, 0xc0003f8b70?}, 0xc0003b9880)
github.com/docker/distribution/registry/handlers/app.go:324 +0x1c0d
github.com/docker/distribution/registry.NewRegistry({0xe5e8a0?, 0xc0003f81b0?}, 0xc0003b9880)
github.com/docker/distribution/registry/registry.go:161 +0x145
github.com/docker/distribution/registry.glob..func1(0xc0000341a0?, {0xc0003096e0, 0x1, 0x1})
github.com/docker/distribution/registry/registry.go:119 +0x18f
github.com/docker/distribution/vendor/github.com/spf13/cobra.(*Command).execute(0x1369f60, {0xc0003096a0?, 0x2?, 0x2?})
github.com/docker/distribution/vendor/github.com/spf13/cobra/command.go:495 +0x44f
github.com/docker/distribution/vendor/github.com/spf13/cobra.(*Command).Execute(0x136a100)
github.com/docker/distribution/vendor/github.com/spf13/cobra/command.go:560 +0xaa
main.main()
github.com/docker/distribution/cmd/registry/main.go:23 +0x25
time="2023-09-04T13:45:50.365608691Z" level=warning msg="No HTTP secret provided - generated random secret. This may cause problems with uploads if multiple registries are behind a load-balancer. To provide a shared secret, fill in http.secret in the configuration file or set the REGISTRY_HTTP_SECRET environment variable." go.version=go1.19.9 instance.id=48510656-5470-4fc4-91fe-001413ab6c29 service=registry version=2.8.2
time="2023-09-04T13:45:50.36574565Z" level=info msg="redis not configured" go.version=go1.19.9 instance.id=48510656-5470-4fc4-91fe-001413ab6c29 service=registry version=2.8.2
time="2023-09-04T13:45:50.365800129Z" level=info msg="Starting upload purge in 29m0s" go.version=go1.19.9 instance.id=48510656-5470-4fc4-91fe-001413ab6c29 service=registry version=2.8.2
time="2023-09-04T13:45:50.366012801Z" level=info msg="using inmemory blob descriptor cache" go.version=go1.19.9 instance.id=48510656-5470-4fc4-91fe-001413ab6c29 service=registry version=2.8.2
time="2023-09-04T13:45:50.366264921Z" level=info msg="Starting cached object TTL expiration scheduler..." go.version=go1.19.9 instance.id=48510656-5470-4fc4-91fe-001413ab6c29 service=registry version=2.8.2
panic: Get "https://registry-1.docker.io/v2/": dial tcp: lookup registry-1.docker.io on 10.1.3.1:53: server misbehaving

goroutine 1 [running]:
github.com/docker/distribution/registry/handlers.NewApp({0xe5e8a0?, 0xc0003cb380?}, 0xc0003b0a80)
github.com/docker/distribution/registry/handlers/app.go:324 +0x1c0d
github.com/docker/distribution/registry.NewRegistry({0xe5e8a0?, 0xc0003ca9c0?}, 0xc0003b0a80)
github.com/docker/distribution/registry/registry.go:161 +0x145
github.com/docker/distribution/registry.glob..func1(0xc0000341a0?, {0xc00034d2c0, 0x1, 0x1})
github.com/docker/distribution/registry/registry.go:119 +0x18f
github.com/docker/distribution/vendor/github.com/spf13/cobra.(*Command).execute(0x1369f60, {0xc00034d280?, 0x2?, 0x2?})
github.com/docker/distribution/vendor/github.com/spf13/cobra/command.go:495 +0x44f
github.com/docker/distribution/vendor/github.com/spf13/cobra.(*Command).Execute(0x136a100)
github.com/docker/distribution/vendor/github.com/spf13/cobra/command.go:560 +0xaa
main.main()
github.com/docker/distribution/cmd/registry/main.go:23 +0x25
time="2023-09-04T13:46:51.118164601Z" level=warning msg="No HTTP secret provided - generated random secret. This may cause problems with uploads if multiple registries are behind a load-balancer. To provide a shared secret, fill in http.secret in the configuration file or set the REGISTRY_HTTP_SECRET environment variable." go.version=go1.19.9 instance.id=234aab86-a040-41c0-b5e4-f3298e494772 service=registry version=2.8.2
time="2023-09-04T13:46:51.11879609Z" level=info msg="redis not configured" go.version=go1.19.9 instance.id=234aab86-a040-41c0-b5e4-f3298e494772 service=registry version=2.8.2
time="2023-09-04T13:46:51.119236Z" level=info msg="using inmemory blob descriptor cache" go.version=go1.19.9 instance.id=234aab86-a040-41c0-b5e4-f3298e494772 service=registry version=2.8.2
time="2023-09-04T13:46:51.119742697Z" level=info msg="Starting cached object TTL expiration scheduler..." go.version=go1.19.9 instance.id=234aab86-a040-41c0-b5e4-f3298e494772 service=registry version=2.8.2
time="2023-09-04T13:46:51.120185025Z" level=info msg="Starting upload purge in 16m0s" go.version=go1.19.9 instance.id=234aab86-a040-41c0-b5e4-f3298e494772 service=registry version=2.8.2
panic: Get "https://registry-1.docker.io/v2/": dial tcp: lookup registry-1.docker.io on 10.1.3.1:53: server misbehaving

goroutine 1 [running]:
github.com/docker/distribution/registry/handlers.NewApp({0xe5e8a0?, 0xc0003f4b70?}, 0xc0003b7880)
github.com/docker/distribution/registry/handlers/app.go:324 +0x1c0d
github.com/docker/distribution/registry.NewRegistry({0xe5e8a0?, 0xc0003f41b0?}, 0xc0003b7880)
github.com/docker/distribution/registry/registry.go:161 +0x145
github.com/docker/distribution/registry.glob..func1(0xc0000341a0?, {0xc000083700, 0x1, 0x1})
github.com/docker/distribution/registry/registry.go:119 +0x18f
github.com/docker/distribution/vendor/github.com/spf13/cobra.(*Command).execute(0x1369f60, {0xc0000836c0?, 0x2?, 0x2?})
github.com/docker/distribution/vendor/github.com/spf13/cobra/command.go:495 +0x44f
github.com/docker/distribution/vendor/github.com/spf13/cobra.(*Command).Execute(0x136a100)
github.com/docker/distribution/vendor/github.com/spf13/cobra/command.go:560 +0xaa
main.main()
github.com/docker/distribution/cmd/registry/main.go:23 +0x25

---------------------------------------------------

**here is the docker output**
root@dockerregistry:~# docker ps
**CONTAINER ID IMAGE COMMAND CREATED STATUS PORTS NAMES**

d5de20eaa1dd registry:latest "/entrypoint.sh /etc…" 24 minutes ago Restarting (2) 9 seconds ago

**Comment by mikef-phy** [comment]
`goroutine 1 [running]:
github.com/docker/distribution/registry/handlers.NewApp({0xe5e8a0?, 0xc0003f8b70?}, 0xc0003b9880)
	github.com/docker/distribution/registry/handlers/app.go:324 +0x1c0d
github.com/docker/distribution/registry.NewRegistry({0xe5e8a0?, 0xc0003f81b0?}, 0xc0003b9880)
	github.com/docker/distribution/registry/registry.go:161 +0x145
github.com/docker/distribution/registry.glob..func1(0xc0000341a0?, {0xc0003096e0, 0x1, 0x1})
	github.com/docker/distribution/registry/registry.go:119 +0x18f
github.com/docker/distribution/vendor/github.com/spf13/cobra.(*Command).execute(0x1369f60, {0xc0003096a0?, 0x2?, 0x2?})
	github.com/docker/distribution/vendor/github.com/spf13/cobra/command.go:495 +0x44f
github.com/docker/distribution/vendor/github.com/spf13/cobra.(*Command).Execute(0x136a100)
	github.com/docker/distribution/vendor/github.com/spf13/cobra/command.go:560 +0xaa
main.main()
	github.com/docker/distribution/cmd/registry/main.go:23 +0x25
time="2023-09-04T13:45:50.365608691Z" level=warning msg="No HTTP secret provided - generated random secret. This may cause problems with uploads if multiple registries are behind a load-balancer. To provide a shared secret, fill in http.secret in the configuration file or set the REGISTRY_HTTP_SECRET environment variable." go.version=go1.19.9 instance.id=48510656-5470-4fc4-91fe-001413ab6c29 service=registry version=2.8.2 
time="2023-09-04T13:45:50.36574565Z" level=info msg="redis not configured" go.version=go1.19.9 instance.id=48510656-5470-4fc4-91fe-001413ab6c29 service=registry version=2.8.2 
time="2023-09-04T13:45:50.365800129Z" level=info msg="Starting upload purge in 29m0s" go.version=go1.19.9 instance.id=48510656-5470-4fc4-91fe-001413ab6c29 service=registry version=2.8.2 
time="2023-09-04T13:45:50.366012801Z" level=info msg="using inmemory blob descriptor cache" go.version=go1.19.9 instance.id=48510656-5470-4fc4-91fe-001413ab6c29 service=registry version=2.8.2 
time="2023-09-04T13:45:50.366264921Z" level=info msg="Starting cached object TTL expiration scheduler..." go.version=go1.19.9 instance.id=48510656-5470-4fc4-91fe-001413ab6c29 service=registry version=2.8.2 
panic: Get "https://registry-1.docker.io/v2/": dial tcp: lookup registry-1.docker.io on 10.1.3.1:53: server misbehaving

goroutine 1 [running]:
github.com/docker/distribution/registry/handlers.NewApp({0xe5e8a0?, 0xc0003cb380?}, 0xc0003b0a80)
	github.com/docker/distribution/registry/handlers/app.go:324 +0x1c0d
github.com/docker/distribution/registry.NewRegistry({0xe5e8a0?, 0xc0003ca9c0?}, 0xc0003b0a80)
	github.com/docker/distribution/registry/registry.go:161 +0x145
github.com/docker/distribution/registry.glob..func1(0xc0000341a0?, {0xc00034d2c0, 0x1, 0x1})
	github.com/docker/distribution/registry/registry.go:119 +0x18f
github.com/docker/distribution/vendor/github.com/spf13/cobra.(*Command).execute(0x1369f60, {0xc00034d280?, 0x2?, 0x2?})
	github.com/docker/distribution/vendor/github.com/spf13/cobra/command.go:495 +0x44f
github.com/docker/distribution/vendor/github.com/spf13/cobra.(*Command).Execute(0x136a100)
	github.com/docker/distribution/vendor/github.com/spf13/cobra/command.go:560 +0xaa
main.main()
	github.com/docker/distribution/cmd/registry/main.go:23 +0x25
time="2023-09-04T13:46:51.118164601Z" level=warning msg="No HTTP secret provided - generated random secret. This may cause problems with uploads if multiple registries are behind a load-balancer. To provide a shared secret, fill in http.secret in the configuration file or set the REGISTRY_HTTP_SECRET environment variable." go.version=go1.19.9 instance.id=234aab86-a040-41c0-b5e4-f3298e494772 service=registry version=2.8.2 
time="2023-09-04T13:46:51.11879609Z" level=info msg="redis not configured" go.version=go1.19.9 instance.id=234aab86-a040-41c0-b5e4-f3298e494772 service=registry version=2.8.2 
time="2023-09-04T13:46:51.119236Z" level=info msg="using inmemory blob descriptor cache" go.version=go1.19.9 instance.id=234aab86-a040-41c0-b5e4-f3298e494772 service=registry version=2.8.2 
time="2023-09-04T13:46:51.119742697Z" level=info msg="Starting cached object TTL expiration scheduler..." go.version=go1.19.9 instance.id=234aab86-a040-41c0-b5e4-f3298e494772 service=registry version=2.8.2 
time="2023-09-04T13:46:51.120185025Z" level=info msg="Starting upload purge in 16m0s" go.version=go1.19.9 instance.id=234aab86-a040-41c0-b5e4-f3298e494772 service=registry version=2.8.2 
panic: Get "https://registry-1.docker.io/v2/": dial tcp: lookup registry-1.docker.io on 10.1.3.1:53: server misbehaving`

**Comment by mikef-phy** [comment]
as soon as you get access to docker registry it comes online.  That log just keeps repeating as it restarts over and over.  If you want to see the behavior just sever connectivity to docker.io and reboot.

**Comment by mikef-phy** [comment]
panic: Get "https://registry-1.docker.io/v2/": dial tcp: lookup registry-1.docker.io on 10.1.3.1:53: server misbehaving
